log2/logger.go

```diff
@@ -0,0 +1,185 @@
+package log2
+
+import (
+	"io"
+	"log"
+	"strings"
+	"sync"
+)
+
+// Level represents the logging level.
+type Level uint32
+
+const (
+	// NotSet level is used to indicate that no level has been set
+	// and allow for a default to be used
+	NotSet Level = iota
+
+	// Off is intended to avoid tracing any action.
+	Off
+
+	// Fatal
+	Fatal
+
+	// Error
+	Error
+
+	// Warn
+	Warn
+
+	// Info
+	Info
+
+	// Debug
+	Debug
+
+	// Trace
+	Trace
+)
+
+// LevelFromString returns a Level type for the named log level, or
+// "NotSet" if the level passed as argument is invalid.
+func LevelFromString(level string) Level {
+	level = strings.ToLower(strings.TrimSpace(level))
+	switch level {
+	case "off":
+		return Off
+	case "fatal":
+		return Fatal
+	case "error":
+		return Error
+	case "warn":
+		return Warn
+	case "info":
+		return Info
+	case "debug":
+		return Debug
+	case "trace":
+		return Trace
+	default:
+		return NotSet
+	}
+}
+
+type Logger interface {
```

Comment:
We need to document each method to explicitly say when it is correct to use one or another level. If we don't. expect even more incorrect logging than now. I would put it here on the interface. Al least a summary of it. The document design can have the tl;dr

Reply:
We need to point out also that we assume the log reader will filter out the unwanted information when using the verbose levels (trace, debug). Having levels for the purpose of filtering out information dynamically, and also requiring the log reader to filter it looks a bit weird.